velox/connectors/Connector.h

```diff
@@ -64,6 +64,10 @@ struct ConnectorSplit {
   virtual std::string getFileName() const {
     return "";
   }
+
+  virtual std::string getFullFileName() const {
```

Comment:
Should we call this getAbsoluteFilePath or getFilePath?

Reply:
I'm not sure I follow. Would you elaborate a bit? It sounds like there might be some design problem here. file path is an implementation detail of the HiveConnectorSplit. Why is it leaking into generic connector interface?

Reply:
Because hive connector is depending on velox exec module (through memory reclaimer), and we cannot have exec module depending on hive connector back

Reply:
Not sure if that will work well. With delta files, even Hive splits now start to have many files.

Reply:
`getFullBaseFileName` then?

Reply:
so you are talking about one split may be have multiple files related ?

Reply:
@gaoyangxiaozhu Not all connectors use files in splits and even HiveConnector sometimes has splits that have many files. In general, file is not a common property of a split, hence, it doesn't belong to ConnectorSplit interface. We can replace this with a generic API that returns a string that represents a summary of the split, but it cannot have the semantics of a file. How do we use this API anyway?

Reply:
it's for support spark `input_file_name` function, and in Spark it always assume split should have a related file if not empty will be returned.  `input_file_name` is very useful for debug purpose.

checking this https://github.com/facebookincubator/velox/commit/ee84bcb287eeae2c8908608d60fa4f2480d77bdc

you can also checking the PR detail for how spark does. 

For this part `In general, file is not a common property of a split`, I think if it is ok we do refactor in seperate PR since currently the change only for sparrk `input_file_name` scenario support.

Reply:
any thoughts @mbasmanova ?

Reply:
@gaoyangxiaozhu This design is incorrect. We cannot expose driverCtx->inputFileName property as it is cannot be supported in a generic way. We need a different design. Perhaps, 'input_file_name' function can be re-written into returning a synthetic column like $path or $bucket from TableScan.